packages/suite/src/components/wallet/TransactionItem/CoinjoinBatchItem.tsx

```diff
@@ -163,10 +168,21 @@ export const CoinjoinBatchItem = ({
     isPending,
 }: CoinjoinBatchItemProps) => {
     const lastTx = transactions[0];
-    const amount = sumTransactions(transactions);
-    const fiatAmount = sumTransactionsFiat(transactions, localCurrency);
     const { FiatAmountFormatter } = useFormatters();
-    const isMissingFiatRates = transactions.some(tx => !tx.rates?.[localCurrency]);
+    const historicFiatRates = useSelector(selectHistoricFiatRates);
+    const amount = sumTransactions(transactions);
+    const fiatAmount = sumTransactionsFiat(transactions, localCurrency, historicFiatRates);
+    const isMissingFiatRates = transactions.some(tx => {
+        const fiatRateKey = getFiatRateKey(
+            tx.symbol,
+            localCurrency as FiatCurrencyCode,
+            tx.tokens[0]?.contract as TokenAddress,
+        );
+        const roundedTimestamp = roundTimestampToNearestPastHour(tx.blockTime as Timestamp);
+        const historicRate = historicFiatRates?.[fiatRateKey]?.[roundedTimestamp as Timestamp];
+
+        return !historicRate;
```

Comment:
`historicRate` is a `number|undefined`. `!0` is `true`.

It is good to be strict in not doing implicit type conversions.

Reply:
In case  `0` resolving as `true` is intentional (in sense `isMissingFiatRates === true` then, please leave a comment and/or better to make it absolutely clear by `historicRate === undefined || historicRate === 0`